sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -286,7 +295,11 @@ private HttpRequest configRequest(final HttpRequest request, final SwaggerMethod
                 }
             }
 
-            if (isJson) {
+            if (bodyContentObject instanceof BinaryData) {
+                byte[] body = ((BinaryData) bodyContentObject).toBytes();
+                request.setBody(body);
```

Comment:
`setBody(byte[])` implicitly sets the `Content-Length` header

Reply:
removed `content-length` line